pkg/config/server/server.go

```diff
@@ -328,6 +330,21 @@ type ConfigFileAuthGithub struct {
 	Scopes       []string `mapstructure:"scopes" json:"scopes,omitempty" default:"[\"read:user\", \"user:email\"]"`
 }
 
+type ConfigFileAuthCustom struct {
+	Enabled bool `mapstructure:"enabled" json:"enabled,omitempty" default:"false"`
+
+	ClientID     string `mapstructure:"clientID" json:"clientID,omitempty"`
+	ClientSecret string `mapstructure:"clientSecret" json:"clientSecret,omitempty"`
+
+	AuthorizationURL string `mapstructure:"authorizationURL" json:"authorizationURL,omitempty"`
+	AccessTokenURL   string `mapstructure:"accessTokenURL" json:"accessTokenURL,omitempty"`
+	ResourceURL      string `mapstructure:"resourceURL" json:"resourceURL,omitempty"`
```

Comment:
Is this referring to a userinfo URL? If so, it might be helpful to rename it to `UserInfoURL`